src/actors/generic.py

```diff
@@ -0,0 +1,134 @@
+            self.log = logging.getLogger(self.actor.name)
+
+        # Init the ActorTypeDispatcher
+        super(ActorTypeDispatcher, self).__init__(*args, **kwargs)
+        self.log.info(f"Initialized actor '{self.actor.name}'.")
+
+    def receiveMsg_str(self, message: str, sender: ActorAddress):
+        """Parses incoming messages containing strings."""
+        self.log.debug("Received str %s from sender %s", message, sender)
+
+    def receiveMsg_int(self, message: int, sender: ActorAddress):
+        """Parses incoming messages containing integers."""
+        self.log.debug("Received int %s from sender %s", message, sender)
+
+    def receiveMsg_dict(self, message: dict, sender: ActorAddress):
+        """
+        Parses incoming messages containing dictionaries.
+
+        Make sure you SUPER this method if you are overriding it.
+        """
+        self.log.debug("Received dict %s from sender %s", message, sender)
+
+        # If the dictionary has actor-name keys, assume it is an address book.
+        if list(message.keys())[0] in ActorNames:
+            self.address_book = message
+
+    def receiveMsg_AddressBook(self, message: AddressBook, sender: ActorAddress):
+        """
+        DO NOT OVERRIDE;
+
+        Parses incoming messages containing dictionaries.
+        """
+        # Currently this fails, the arriving object is empty. Needs some work.
+        self.log.debug("Received address book from sender %s", message, sender)
+        self.log.debug("Received Book has keys: %s", message.all().keys())
+        self.address_book = message
+
+    def receiveMsg_WakeupMessage(self, message: dict, sender: ActorAddress):
+        """
+        DO NOT OVERRIDE;
+
+        Continuously runs the code in the looping method.
+        """
+        if self.loop_enabled:
+            self.wakeupAfter(self.loop_period_seconds)
+        self.log.debug("Waking up.")
+        self.loop()
+
+    def receiveMsg_Requests(self, message: Requests, sender: ActorAddress):
+        """
+        DO NOT OVERRIDE;
+
+        Parses incoming messages containing Request enumerations.
+        """
+        self.log.debug("Received enum %s from sender %s", message.name, sender)
+
+        if message is Requests.START:
+            # When the start message is sent to an actor, record the sender as the parent.
+            self.parent = sender
+            self.send(sender, Requests.STARTING)
+            self.start(message, sender)
+        elif message is Requests.STOP:
+            self.send(sender, Requests.STOPPING)
+            self.set_loop_enabled(False)
+            self.stop(message, sender)
+        elif message is Requests.ARE_YOU_ALIVE:
+            self.send(sender, Requests.YES)
+        elif message in Requests:
+            pass
+        else:
+            msg = "Unrecognized Request Enum value sent."
+            self.log.error(msg)
+            raise Exception(msg)
+
+    def set_loop_period_seconds(self, seconds: float):
```

Comment:
next few methods are specific to the sensor, should be moved to there

Reply:
They could be used in more than sensor.

Reply:
could be isnt for sure, lets move it to sensor for now